csgo_internal/csgo_internal/RageBot.cpp

```diff
@@ -158,304 +136,311 @@ void CRageBot::GoToTarget()
 }
 
-int hitboxes[7] =
+int hitboxes[8] =
 {
-	HITBOX_HEAD,
-	HITBOX_CHEST,
 	HITBOX_PELVIS,
-	HITBOX_RIGHT_FOREARM,
-	HITBOX_LEFT_FOREARM,
+	HITBOX_CHEST,
+	HITBOX_RIGHT_UPPER_ARM,
+	HITBOX_RIGHT_THIGH,
+	HITBOX_LEFT_THIGH,
 	HITBOX_RIGHT_CALF,
-	HITBOX_LEFT_CALF
+	HITBOX_LEFT_CALF,
+	HITBOX_HEAD
 };
 
-// goes through the above hitboxes and selects the best point on the target that will deal the most damage
-bool CRageBot::BestAimPointAll( CBaseEntity* target, Vector& hitbox )
-{
-	BestPoint aim_point;
 
-	for( int i = 0; i < 7; i++ )
+
+// goes through the above hitboxes
+// selects the point from the target's hitboxes that will deal the most damage
+bool CRageBot::BestAimPointAll(CBaseEntity* target, Vector& best_point)
+{
+	int highest_damage = 0;
+	for (int i = 0; i < 8; i++)
 	{
-		Hitbox hitbox( hitboxes[ i ] );
-		GetHitbox( target, &hitbox );
+		BestPoint aim_point;
+		Hitbox hitbox(hitboxes[i]);
+		GetHitbox(target, &hitbox);
 
-		if( !GetBestPoint( target, &hitbox, &aim_point ) )
+		if (!GetBestPoint(target, &hitbox, &aim_point)) {
 			continue;
-	}
+		}
 
-	if( aim_point.dmg > Vars.Ragebot.AutoWallDmg )
-	{
-		hitbox = aim_point.point;
-		return true;
+		if (aim_point.dmg > highest_damage) {
+			highest_damage = aim_point.dmg;
+			best_point = aim_point.point;
+		}
 	}
-		
-
-	return false;
+	return(highest_damage > Vars.Ragebot.AutoWallDmg);
 }
 
 // checks multiple points inside the selected hitbox and picks the best point which will deal the most damage
-bool CRageBot::BestAimPointHitbox( CBaseEntity* target, Vector& hitbox )
+bool CRageBot::BestAimPointHitbox(CBaseEntity* target, Vector& best_point)
 {
 	BestPoint aim_point;
 
 	int hitboxnum = 0;
 
-	switch( Vars.Ragebot.Hitbox )
-	{
+	switch (Vars.Ragebot.Hitbox) {
 	case 0:
 		hitboxnum = 3;
+		break;
 	case 1:
 		hitboxnum = 6;
+		break;
 	case 2:
+		break;
 	case 3:
 		hitboxnum = 4;
+		break;
 	case 4:
 		hitboxnum = 7;
+		break;
 	case 5:
 		hitboxnum = 1;
+		break;
 	case 6:
 		hitboxnum = 0;
+		break;
 	default:
 		hitboxnum = 0;
+		break;
 	}
 
-	Hitbox hitboxx( hitboxnum );
-	GetHitbox( target, &hitboxx );
+	// Generate Hitbox
+	Hitbox target_hitbox(hitboxnum);
+	GetHitbox(target, &target_hitbox);
 
-	if( !GetBestPoint( target, &hitboxx, &aim_point ) )
+	//Find best point
+	if (!GetBestPoint(target, &target_hitbox, &aim_point))
 		return false;
 
-	if( aim_point.dmg > Vars.Ragebot.AutoWallDmg )
-	{
-		hitbox = aim_point.point;
+	if (aim_point.dmg > Vars.Ragebot.AutoWallDmg) {
+		best_point = aim_point.point;		
 		return true;
 	}
-
 	return false;
 }
 
-bool CRageBot::EntityIsValid( int i )
+bool CRageBot::EntityIsValid(int i)
 {
-	CBaseEntity* Entity = I::ClientEntList->GetClientEntity( i );
+	CBaseEntity* Entity = I::ClientEntList->GetClientEntity(i);
 
-	if( !Entity )
-		return false;
-	if( Entity == G::LocalPlayer )
+	if (!Entity)
+	{
+		
 		return false;
+	}
 
-	if( !Vars.Ragebot.FriendlyFire )
+	if (Entity->IsEnemy())
```

Comment:
The previous check wasn't needed... this check now disables any friendly fire in the ragebot which is needed for modes like FFA or custom server gamemodes.